doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -525,12 +531,13 @@ counterparts when the relation involved is not Leibniz equality.
 Notice, however, that using the prefixed tactics it is possible to
 pass additional arguments such as ``using relation``.
 
-.. tacv:: setoid_reflexivity
-          setoid_symmetry {? in @ident}
-          setoid_transitivity
-          setoid_rewrite {? @orientation} @term {? at @occurrences} {? in @ident}
-          setoid_replace @term with @term {? using relation @term} {? in @ident} {? by @ltac_expr3}
-   :name: setoid_reflexivity; setoid_symmetry; setoid_transitivity; setoid_rewrite; setoid_replace
+.. tacn:: setoid_reflexivity
+          setoid_symmetry {? in @ident }
+          setoid_transitivity @one_term
+          setoid_rewrite {? {| -> | <- } } @constr_with_bindings {? at @occurrences } {? in @ident }
```

Comment:
What is this `constr_with_bindings` which is used here and in two other places but not defined (listed in the exclusion list in `conf.py`)?

Reply:
I thought that definition belongs in the tactics chapter, so no point adding in this PR.

Reply:
I don't know what this is. Is this just a `@one_term {* @binder }` or something like that? If yes, what about just splicing it?

Reply:
```

constr_with_bindings: [
| one_term OPT ( "with" bindings )
]


bindings: [
| LIST1 simple_binding
| LIST1 one_term
]


simple_binding: [
| "(" ident ":=" term ")"
| "(" num ":=" term ")"
]


```

Reply:
I find the indirection a bit unfortunate for something that is just an optional clause.

Reply:
A lament or a suggestion that I splice `constr_with_bindings`?  Here are the productions that use it (mostly tactics).

```
| OPT ">" constr_with_bindings      (* SSR plugin *)
| "specialize" constr_with_bindings OPT ( "as" simple_intropattern )
| "contradiction" OPT constr_with_bindings
| "substitute" OPT [ "->" | "<-" ] constr_with_bindings
| "setoid_rewrite" OPT [ "->" | "<-" ] constr_with_bindings OPT ( "at" occurrences ) OPT ( "in" ident )
| "setoid_rewrite" OPT [ "->" | "<-" ] constr_with_bindings "in" ident "at" occurrences
| "elim" constr_with_bindings_arg OPT ( "using" constr_with_bindings )
| "eelim" constr_with_bindings_arg OPT ( "using" constr_with_bindings )
| "functional" "induction" term OPT ( "using" constr_with_bindings ) OPT ( "as" simple_intropattern )      (* funind plugin *)
| "soft" "functional" "induction" LIST1 one_term OPT ( "using" constr_with_bindings ) OPT ( "as" simple_intropattern )      (* funind plugin *)
| LIST1 induction_clause SEP "," OPT ( "using" constr_with_bindings ) OPT opt_clause
```

Reply:
A suggestion.